### PR TITLE
[tsp-client] Add support for package-lock.json files

### PR DESCRIPTION
**Pull request by catalinaperalta**
Fixes https://github.com/Azure/azure-sdk-tools/issues/7057

### CODE REVIEW

File: tools/tsp-client/src/index.ts [hunk 1]
```diff
@@ -172,8 +172,18 @@ async function syncTspFiles(outputDir: string, localSpecRepo?: string) {
     await removeDirectory(cloneDir);
   }
 
-  const emitterPath = joinPaths(repoRoot, "eng", "emitter-package.json");
-  await cp(emitterPath, joinPaths(srcDir, "package.json"), { recursive: true });
+  try {
+    const emitterLockPath = joinPaths(repoRoot, "eng", "emitter-package-lock.json");
+    await cp(emitterLockPath, joinPaths(srcDir, "package-lock.json"), { recursive: true });
+  } catch (err) {
+    Logger.debug("emitter-package-lock.json not found, will look for emitter-package.json");
```
Comment (joheredi) [inline tools/tsp-client/src/index.ts:179]: There could be other reasons (maybe not too common) why copy failed, for example if the disk is full or for some reason `srcDir` doesn't exist. If users fall into those situations, this log may be missleading

File: tools/tsp-client/src/index.ts [hunk 2]
```diff
@@ -201,8 +211,19 @@ async function generate({
   const mainFilePath = await discoverMainFile(srcDir);
   const resolvedMainFilePath = joinPaths(srcDir, mainFilePath);
   Logger.info("Installing dependencies from npm...");
-  await installDependencies(srcDir);
-
+  const args: string[] = [];
+  try {
+    // Check is package-lock.json exists, if it does, we'll install dependencies through `npm ci`
```
Comment (joheredi) [inline tools/tsp-client/src/index.ts:216]:
```suggestion
    // Check if package-lock.json exists, if it does, we'll install dependencies through `npm ci`
```

File: tools/tsp-client/src/npm.ts [hunk 3]
```diff
@@ -19,9 +19,9 @@ export async function createPackageJson(rootPath: string, deps: Set<string>): Pr
   await writeFile(filePath, packageJson);
 }
 
-export function installDependencies(workingDir: string): Promise<void> {
+export async function npmInstallationCommand(workingDir: string, args: string[]): Promise<void> {
```
Comment (joheredi) [inline tools/tsp-client/src/npm.ts:22]: Is it fair to say that this function became `npmCommand` with the change of taking args and is no longer scoped to the install command?

Comment (joheredi) [reply tools/tsp-client/src/npm.ts:22]: If so, maybe we should rename

**Review by joheredi — APPROVED** [review]
Left a few minor comments

File: tools/tsp-client/src/index.ts (hunk 1 above)
Comment (catalinaperalta) [reply tools/tsp-client/src/index.ts]: That's true, I updated the log to include the original error but still wont fail in this case if we have an emitter-package.json. If the issue happens with emitter-package.json then we will fail

File: tools/tsp-client/src/npm.ts (hunk 3 above)
Comment (catalinaperalta) [reply tools/tsp-client/src/npm.ts]: Renamed

**Pull request merged**